json/flagd_definitions_test.go

```diff
@@ -2,19 +2,29 @@ package flagd_definitions_test
 
 import (
 	"fmt"
+	"log"
 	"os"
 	"path/filepath"
 	"strings"
 	"testing"
 
 	flagd_definitions "github.com/open-feature/schemas/json"
+
 	"github.com/xeipuuv/gojsonschema"
 )
 
-var schemaLoader gojsonschema.JSONLoader
+var schemaLoader *gojsonschema.SchemaLoader
+var s *gojsonschema.Schema
 
 func init() {
-	schemaLoader = gojsonschema.NewStringLoader(flagd_definitions.FlagdDefinitions)
+	schemaLoader = gojsonschema.NewSchemaLoader()
+	schemaLoader.AddSchemas(gojsonschema.NewStringLoader(flagd_definitions.Targeting))
+	var err error
+	s, err = schemaLoader.Compile(gojsonschema.NewStringLoader(flagd_definitions.FlagdDefinitions))
+	if err != nil {
+		s := fmt.Errorf("err: %v", err)
+		log.Fatal(s)
+	}
```

Comment:
@Kavindu-Dodan you asked about this... this is how we can directly import schemas in Go (notice I import the referenced (`$ref`) schema first, and then the root schema. This means no actual http requests will be made, the schemas are cached and referenced by their `$id`. In most prod systems, this is how it must work - few servers or libraries will actually make HTTP requests for schemas, that's typically more of a development feature (like for IDEs).

See [this link](https://json-schema.org/understanding-json-schema/structuring#schema-identification):

> Even though schemas are identified by URIs, those identifiers are not necessarily network-addressable. They are just identifiers. Generally, implementations don't make HTTP requests (https://) or read from the file system (file://) to fetch schemas. Instead, they provide a way to load schemas into an internal schema database. When a schema is referenced by it's URI identifier, the schema is retrieved from the internal schema database.

Reply:
Note I improved some var-names here since I made this comment.

Reply:
I didn't even consider this but it's definitely good to know.

Reply:
Cool, thank you for the link and the comment. I thought this to be a valid URL hence my doubt :)